src/linear_algebra/clifford_algebra/spin_group.lean

```diff
@@ -0,0 +1,238 @@
+/-
+Copyright (c) 2022 Jiale Miao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiale Miao
+-/
+import group_theory.group_action.conj_act
+import algebra.star.unitary
+import linear_algebra.clifford_algebra.star
+import linear_algebra.clifford_algebra.even
+
+/-!
+# The Pin group and the Spin group
+
+In this file we define `lipschitz` consisting of all the elements in `(clifford_algebra)ˣ` closed
+under conjugation and construct `pin_group` based on `lipschitz` and `unitary`, and we construct
+`spin_group` based on `pin_group` and `clifford_algebra.even`. Also, we show `pin_group` and
+`spin_group` form a group where the inverse is `star`.
+
+## Main definitions
+
+* `lipschitz`: the Lipschitz group with a quadratic form.
+* `pin_group`: the Pin group with a quadratic form.
+* `spin_group`: the Spin group with a quadratic form.
+
+-/
+
+variables {R : Type*} [field R]
+variables {M : Type*} [add_comm_group M] [module R M]
+variables {Q : quadratic_form R M}
+
+section pin
+open clifford_algebra mul_action
+open_locale pointwise
+
+/--
+`lipschitz Q` is the subgroup of `(clifford_algebra Q)ˣ` consisting of all the elements closed under
+conjugation. See `mem_lipschitz_iff`.
+-/
+def lipschitz (Q : quadratic_form R M): subgroup (clifford_algebra Q)ˣ :=
+(stabilizer (conj_act (clifford_algebra Q)ˣ) (ι Q).range).comap
+  (conj_act.to_conj_act : (clifford_algebra Q)ˣ ≃* conj_act (clifford_algebra Q)ˣ)
+
+lemma mem_lipschitz_iff {x : (clifford_algebra Q)ˣ} : x ∈ lipschitz Q
+  ↔ conj_act.to_conj_act x • (ι Q).range = (ι Q).range := iff.rfl
```

Comment:
A little easier to read as
```suggestion
lemma mem_lipschitz_iff {x : (clifford_algebra Q)ˣ} :
  x ∈ lipschitz Q ↔ conj_act.to_conj_act x • (ι Q).range = (ι Q).range := iff.rfl
```
Generally we try not to wrap in the middle of a statement if psosible.

Reply:
ok, let me check this again